### PR TITLE
Display solved puzzles in profile/settings screen

### LINKED ISSUE #368: Display the list of solved puzzles on the user's profile

### PR DESCRIPTION
**Pull request by matt989253**
Closes #368

### CODE REVIEW

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulHomeTest.kt [hunk 1]
```diff
@@ -592,4 +592,82 @@ class StatefulHomeTest {
...
+                user = user,
+                controller = controller,
+            )
+          }
+        }
+
+    rule.onNodeWithText(strings.sectionSettings).performClick()
+    rule.onNodeWithText(strings.profilePuzzle).performClick()
+    rule.onNodeWithText("00008", substring = true).performClick()
+    rule.onNodeWithText(strings.puzzleNumber("00008")).assertExists()
```
Comment (barmettlerl) [inline mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulHomeTest.kt:629]: You could use here the helper method `rule.setContentWithTestEnvironment()` to simplify the test a bit

**Review by barmettlerl — APPROVED** [review]

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulHomeTest.kt [hunk 2]
```diff
@@ -592,4 +592,82 @@ class StatefulHomeTest {
...
+  @Test
+  fun given_settingsScreen_when_clickingOnPuzzle_then_ItIsOpened() = runTest {
+    val auth = buildAuth { user("email@example.org", "password", "1") }
+    val store = buildStore {
+      collection("users") { document("1", ProfileDocument(solvedPuzzles = listOf("00008"))) }
+    }
+    val assets =
+        FakeAssetManager(
+            "PuzzleId,FEN,Moves,Rating,RatingDeviation,Popularity,NbPlays,Themes,GameUrl\n" +
+                "00008,r6k/pp2r2p/4Rp1Q/3p4/8/1N1P2R1/PqP2bPP/7K b - - 0 24,f2g3 e6e7 b2b1 b3c1 b1c1 h6c1,1852,74,97,1444,crushing hangingPiece long middlegame,https://lichess.org/787zsVup/black#48\n")
```
Comment (KurohanaJuri) [inline mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulHomeTest.kt:605]: I see that you pass the same asset multiple time, maybe you can create a class that extend `FakeAssetManager` and return this string when `readText` is called. 

You can create different class depending on the expected behaviour.

We have done this for the [speech recognizer](https://github.com/epfl-SDP/android/tree/main/mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/infrastructure/speech)

Comment (matt989253) [reply mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulHomeTest.kt]: Fixed in https://github.com/epfl-SDP/android/pull/390/commits/51ac3db25c9965fd4f2656f2fe6a204af68a03b1

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulHomeTest.kt [hunk 3]
```diff
@@ -592,4 +592,82 @@ class StatefulHomeTest {
     rule.onNodeWithText(strings.tournamentsBadgeJoin).assertIsDisplayed()
     rule.onNodeWithText("Tournament 1", ignoreCase = true).assertIsDisplayed()
   }
+
+  @Test
+  fun given_settingsScreen_when_clickingOnPuzzle_then_ItIsOpened() = runTest {
```
[the remark this answers is not in the document]
Comment (matt989253) [reply mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulHomeTest.kt:569]: Fixed in https://github.com/epfl-SDP/android/pull/390/commits/34e52e8ced69407f3e39a6e04a81efd6238af6c5

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulHomeTest.kt (hunk 1 above)
Comment (matt989253) [reply mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulHomeTest.kt]: It was a bit more work than I would have liked, but you're right!
Fixed in https://github.com/epfl-SDP/android/pull/390/commits/2e1f60ffb0fd95e8f23d447428c77c64a88d1364

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt [hunk 4]
```diff
@@ -81,8 +83,11 @@ data class TestEnvironment(
  * information.
  */
 suspend fun ComposeContentTestRule.setContentWithTestEnvironment(
-    store: Store = emptyStore(),
-    auth: Auth = emptyAuth(),
+    userId: String = DefaultId,
```
Comment (alexandrepiveteau) [inline mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt:86]: Missing documentation for the parameter.

```kotlin
* @param userId the user identifier for the test environment user.
```

**Review by alexandrepiveteau — DISMISSED** [review]
Looking great !

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt (hunk 4 above)
Comment (matt989253) [reply mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt:87]: Fixed in https://github.com/epfl-SDP/android/pull/390/commits/c4590eaf815cf86e4d3587ab8be2e0e7e6cc2ede

**Review by alexandrepiveteau — DISMISSED** [review]

**Review by alexandrepiveteau — APPROVED** [review]

**Review by BadrTad — APPROVED** [review]

**Pull request merged**